### PR TITLE
[SPARK-21343] Refine the document for spark.reducer.maxReqSizeShuffleToMem.

### PR DESCRIPTION
**Pull request by jinxing64**
## What changes were proposed in this pull request?

In current code, reducer can break the old shuffle service when `spark.reducer.maxReqSizeShuffleToMem` is enabled. Let's refine document.

### CODE REVIEW

**Comment by jinxing64** [comment]
I didn't include this config in configuration.md. Do I need to?
cc @zsxwing @cloud-fan

**Comment by zsxwing** [comment]
@jinxing64 yeah, please also update configuration.md.

File: core/src/main/scala/org/apache/spark/internal/config/package.scala [hunk 1]
```diff
@@ -323,9 +323,10 @@ package object config {
 
   private[spark] val REDUCER_MAX_REQ_SIZE_SHUFFLE_TO_MEM =
     ConfigBuilder("spark.reducer.maxReqSizeShuffleToMem")
-      .internal()
       .doc("The blocks of a shuffle request will be fetched to disk when size of the request is " +
-        "above this threshold. This is to avoid a giant request takes too much memory.")
+        "above this threshold. This is to avoid a giant request takes too much memory. Note that" +
+        " it's necessary to have the support of shuffle service(at least Spark-2.2) when enable " +
```
Comment (cloud-fan) [inline core/src/main/scala/org/apache/spark/internal/config/package.scala]: is it possible to implement this feature without shuffle service?

File: docs/configuration.md [hunk 2]
```diff
@@ -529,6 +529,15 @@ Apart from these, the following properties are also available, and may be useful
   </td>
 </tr>
 <tr>
+  <td><code>spark.reducer.maxReqSizeShuffleToMem</code></td>
+  <td>200m</td>
+  <td>
+    The blocks of a shuffle request will be fetched to disk when size of the request is above
+    this threshold. This is to avoid a giant request takes too much memory. Note that it's
+    necessary to have the support of shuffle service(at least Spark-2.2) when enable this config.
```
Comment (cloud-fan) [inline docs/configuration.md]: let's also mention how to enable this config, e.g. set it to `200m`

File: docs/configuration.md [hunk 3]
```diff
@@ -529,6 +529,15 @@ Apart from these, the following properties are also available, and may be useful
   </td>
 </tr>
 <tr>
+  <td><code>spark.reducer.maxReqSizeShuffleToMem</code></td>
+  <td>200m</td>
```
Comment (cloud-fan) [inline docs/configuration.md]: this is wrong

**Comment by cloud-fan** [comment]
retest this please

**Comment by cloud-fan** [comment]
LGTM

**Comment by cloud-fan** [comment]
thanks, merging to master/2.2!

**Pull request closed without merging**